packages/starlight/utils/i18n.ts

```diff
@@ -161,6 +169,23 @@ function getLocaleInfo(lang: string) {
 	}
 }
 
+/**
+ * Returns the direction of the passed locale.
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getTextInfo
+ */
+function getLocaleDir(locale: Intl.Locale): 'ltr' | 'rtl' {
+	if ('textInfo' in locale) {
+		// @ts-expect-error - `textInfo` is typed but is available in v8 based environments.
+		return locale.textInfo.direction;
+	} else if ('getTextInfo' in locale) {
+		// @ts-expect-error - `getTextInfo` is typed but is available in some non-v8 based environments.
```

Comment:
Here too? Now I’m doubting what you meant, but I think it’s this?

```suggestion
		// @ts-expect-error - `getTextInfo` is typed but is only available in some non-v8 based environments.
```